source/content/guides/drupal-9-migration/05-drupal-9-to-pantheon.md

```diff
@@ -0,0 +1,377 @@
+---
+title: Migrate to Drupal 9 on Pantheon
+subtitle: Migrate a Drupal 9 Site to Pantheon
+description: Migrate a Composer-managed Drupal 9 Site from another platform to Pantheon.
+categories: [develop]
+cms: drupal-9
+tags: [code, launch, migrate, site, updates]
+reviewed: "2021-11-12"
+layout: guide
+showtoc: true
+permalink: docs/guides/drupal-9-migration/drupal-9-to-pantheon
+anchorid: drupal-9-migration/drupal-9-to-pantheon
+editpath: drupal-9-migration/05-drupal-9-to-pantheon.md
+---
+
+This doc shows how to migrate an existing Composer-managed Drupal 9 site from another platform to a new Drupal 9 site with [Integrated Composer](/integrated-composer) on Pantheon.
+
+## Overview
+
+In this migration you'll do the following:
+
+- Using Git, clone your existing site locally
+- Create a new Drupal 9 development site on Pantheon, and clone it locally
+
+Working locally across both the existing and new site codebases, you'll:
+
+- Copy selected portions of the existing site code into the new site code
+- Copy portions of `composer.json` from the existing site to the new site
+- Copy and commit the existing site's custom modules, themes, profiles, and other dependencies to the new site
+- Make any needed customizations to `settings.php` on the new site
+- Carry over any additional code customizations from the existing site to the new Pantheon Drupal 9 site code
+
+<Alert title="Integrated Composer and existing files" type="danger">
+
+Integrated Composer will break if `composer install` tries to modify any files that are committed in Git. To avoid this, [DO THIS THING *****]
```

Comment:
what are the steps we should offer folks here to avoid having `composer install` modify files that are committed in Git?

Reply:
I think it's included in L124

Reply:
Kevin is correct, it's addressed there, [item #3 under "Composer packages"](https://pr-6862--pantheon-docs.my.pantheonfrontend.website/docs/guides/drupal-9-migration/drupal-9-to-pantheon#composer-packages):

> `git status` to see if files have been added that aren't ignored by `.gitignore`. If anything shows up other than `composer.*`, add it to `.gitignore` until `git status` only shows the composer files being modified.

To avoid it, run `composer install` locally and run `git status` to make sure nothing has been changed by doing that.  If their site doesn't do anything special/unusual -- they just install contrib modules and add their own custom code, it shouldn't happen... but if, for example, people add their own composer scripts--something the document never describes doing, but people sometimes do anyway--it can cause this to occur. 

The inclusion of this point in this document and the "Convert Custom Upstream to Integrated Composer" document is a question of balancing completeness and clarity for people with complex situations, against simplicity for people who have simple situations.  People with simple situations won't need to be told this because it won't happen to them.

Reply:
I think edits have moved this to L113 / item #2. The text in question:

> `git status` to see if files have been added that aren't ignored by `.gitignore`. If anything shows up other 
> than `composer.*`, add it to `.gitignore` until `git status` only shows the composer files being modified.